spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -121,20 +121,30 @@ public Blob getGoogleStorageObject() throws IOException {
 	}
 
 	/**
-	 * Creates a signed URL to an object if it exists. This method will fail if this storage resource
-	 * was not created using service account credentials.
+	 * Creates a signed URL to an object.
+	 * <p>If an object doesn't exist and {@code createBlobIfNotExists} is {@code true}, then the
+	 * object is created in GCS.
+	 * <p>This method will fail if this storage resource was not created using service account
+	 * credentials.
 	 * @param timeUnit the time unit used to determine how long the URL is valid.
 	 * @param timePeriods the number of periods to determine how long the URL is valid.
 	 * @return the URL if the object exists, and null if it does not.
 	 * @throws IOException
 	 */
 	public URL createSignedUrl(TimeUnit timeUnit, long timePeriods) throws IOException {
 		Blob blob = this.getGoogleStorageObject();
+
 		if (blob == null) {
-			return null;
+			if (!this.createBlobIfNotExists) {
+				return null;
+			}
+
+			createBlob();
```

Comment:
A question here is whether this method should create the object if it doesn't exist, or if `GoogleStorageResourceBucket.createRelative()` should create the object explicitly.

Reply:
I think we should let the functions inside the resource do the creating (as it is done now). 

The reason is the createBlob() function is private. If we were to make createBlob() public then that kinda sabotages the auto-create setting.

Reply:
Currently, we only create files on `getOutputStream()`. This is equivalent to saying that we only create a file if you want to write to it. But a file doesn't have to be written to in order to be created.

So I think that this follows your rule: have the API methods create the file in GCS so we don't need to expose `createBlob()`?